### ISSUE TITLE
[PM-221] several items found when running on periodograms

### ISSUE THREAD

**Issue by mayani** [pegasus-isi/pegasus#341]
I ran the pegasus-statistics on my periodograms (very simple jobs using Condor-I/O for staging) of ~1600 clustered jobs (~33k tasks total). 

[1] I don't see the number of tasks in the workflow. 
[2] When taking a long time, it should have some form of ("I am alive") output, either dots or a rotating bar or something. 
[3] The breakdown.txt file created was empty, but the job statistics file had all interesting things.
[4] The computation of the Q length appears to be wrong. 
[5] The stage-out jobs don't use kickstart, but the output in jobs claims kickstart time. We think it should appear in Condor time. 

I will attach the submit directory as compressed tarball.

As a feature request, it would be useful in my case to have an extra statistics output file that shows hosts and jobs on hosts. In the attached case, the site is always "futuregrid" for compute jobs, but there are actually up to 3 different sites involved. Having a file ordered by the hostaddr from the kickstart record might help me to come up with more statistics, e.g. I can later cluster them by some netmask that I know.

**Reporter:** @voeckler
**Resolution:** Fixed
**Watchers:**
@vahi
**Attachments:**

**Comment by mayani** [comment]
**Author:** @vahi

pegasus-statistics has a -c option to designate a pure condor run. that will result in a non empty breakdown.txt file

**Comment by mayani** [comment]
**Author:** @voeckler

Due to bug #342 I don't have enough information for "pure Condor" -- the common log is a goner.

**Comment by mayani** [comment]
**Author:** @vahi

I dont think common log is used by pegasus-statistics . did you try the -c option?

**Comment by mayani** [comment]
**Author:** Prasanth Thomas (prasanth)

Please find the comments for the issues below. 
[1] Currently the number of tasks are not calculated. I am adding it . 
[2] I am adding more debug statements on the status . 
[3] Currently the genstats-breakdown script looks only for the kickstart out files with pattern .XXX ("*.out.[0-9]*) . Will change the pattern (*.out*), so all the cases will be handled. 
[4] If it is pure condor run pegasus-statistics should be called with "c" option. Currently the condorQLen is calculated by incrementing condorQLen whenever a SUBMIT event is encountered in the jobstate log file. And condorQLen is decremented whenever a EXECUTE events comes for 'pure condor run' or 'GRID_SUBMIT' event. This operation is done for jobs whose site is not local.
 Another issue we found is the condorQLen gets decremented every time a EXECUTE EVENT occurs .This was causing a negative value in case pasted below where job gets disconnected and executed again. 
1285354947 merge_periodogram_ID611 SUBMIT 14321.0 futuregrid - 747
1285365858 merge_periodogram_ID611 EXECUTE 14321.0 futuregrid - 747
1285365869 merge_periodogram_ID611 IMAGE_SIZE 14321.0 futuregrid - 747
1285365966 merge_periodogram_ID611 JOB_DISCONNECTED 14321.0 futuregrid - 747
1285365990 merge_periodogram_ID611 JOB_RECONNECTED 14321.0 futuregrid - 747
1285366000 merge_periodogram_ID611 JOB_DISCONNECTED 14321.0 futuregrid - 747
1285366000 merge_periodogram_ID611 JOB_RECONNECT_FAILED 14321.0 futuregrid - 747
1285366040 merge_periodogram_ID611 EXECUTE 14321.0 futuregrid - 747
1285366174 merge_periodogram_ID611 JOB_TERMINATED 14321.0 futuregrid - 747
1285366174 merge_periodogram_ID611 JOB_SUCCESS 0 futuregrid - 747

One solution to fix is use to count the execute only once for a particular job. [
5] For local jobs kickstart was calculated from jobstate information . I am fixing it now. If it is pure condor run pegasus-statistics should be called with "c" option. 

For the feature request, we will add on the python version.
